src/modules/sharing/sharing.service.ts

```diff
@@ -234,6 +242,17 @@ export class SharingService {
       throw new ForbiddenException();
     }
 
+    if (sharing.isProtected() && !plainPassword) {
+      throw new PasswordNeededError();
+    }
+
+    if (sharing.isProtected()) {
+      const decryptedPassword = aes.decrypt(sharing.encryptedPassword, code);
+      if (decryptedPassword !== plainPassword) {
```

Comment:
Do not provide a message for this kind of situation, mix it with other Forbidden Exceptions, so we do not provide context to a brute-force attacker, even if we are using aes256 and a strong key, it is always a good practice to take into count.